app/assets/javascripts/channels/desks.js

```diff
@@ -10,6 +10,34 @@ App.cable = ActionCable.createConsumer();
 
 App.messages = App.cable.subscriptions.create('DesksChannel', {
   received: function(data) {
-    $("#desk_state_" + data.id)[0].innerHTML = data.occupied;
+
+      OCCUPIED_CLASS = 'btn-outline-danger';
+      NOT_OCCUPIED_CLASS = 'btn-outline-success';
+
+      // Get desk state button
+      var target = $("#desk_state_" + data.id)[0];
+
+      if(!target)
+          return;
+
+      // Remove any pre-existing occupied/not_occupied classes before setting new class
+      if (target.classList.contains(OCCUPIED_CLASS))
+          target.classList.remove(OCCUPIED_CLASS);
+
+      if (target.classList.contains(NOT_OCCUPIED_CLASS))
+          target.classList.remove(NOT_OCCUPIED_CLASS);
+
+      // Change button based on whether desk is occupied
+      if(data.occupied)
+      {
+          target.classList.add(OCCUPIED_CLASS);
```

Comment:
Why not remove and add the class in the same place? So that you don't have to check things twice?

Reply:
Turns out I can just do classList.remove without checking!

Reply:
I think this whole function could be a lot more concise. It's quite long given that it doesn't do much.

Reply:
-> Gets Element
-> Removes old class
-> Checks if occupied
-> Adds new class / sets text

I made it as concise as I could (to the extent of my knowledge), but if you have any specific suggestions I'd be more than happy to implement them

Reply:
I have an idea. Check out the next commit